error.go

```diff
@@ -0,0 +1,55 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+
+	"go.uber.org/multierr"
+)
+
+type Error struct {
+	Field string
+	Msg   string
+	Err   error
+}
+
+func (s *Error) Error() string {
+	msg := s.Msg
+	if msg == "" && s.Err != nil {
+		msg = s.Err.Error()
+	}
+
+	if msg == "" {
+		msg = "unknown error"
+	}
+
+	if s.Field == "" {
+		return msg
+	}
+
+	return fmt.Sprintf("%s: %s", s.Field, msg)
+}
+
+func (s *Error) Is(target error) bool {
+	return errors.Is(s.Err, target)
+}
+
+func (s *Error) Unwrap() error {
+	return s.Err
+}
+
+func Append(errs error, err error) error {
```

Comment:
GoDoc style comments highlighting the usage difference between Append, AppendError and AppendFieldError might be nice.

Reply:
Oh yeah, I knew I did forget to add doc comments to these methods... lol